src/private_broadcast.cpp

```diff
@@ -15,10 +16,79 @@ bool PrivateBroadcast::Add(const CTransactionRef& tx) EXCLUSIVE_LOCKS_REQUIRED(!
     return inserted;
 }
 
+std::optional<CTransactionRef> PrivateBroadcast::GetTxForBroadcast() EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    if (m_by_priority.empty()) {
+        return std::nullopt;
+    }
+    const Txid& txid = m_by_priority.begin()->second;
+    auto it = m_by_txid.find(txid);
+    if (Assume(it != m_by_txid.end())) {
+        return it->second.tx;
+    }
+    m_by_priority.erase(m_by_priority.begin());
+    return std::nullopt;
+}
+
+void PrivateBroadcast::PushedToNode(const NodeId& nodeid, const Txid& txid) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
+{
+    LOCK(m_mutex);
+    m_by_nodeid.emplace(nodeid, txid);
+}
+
+bool PrivateBroadcast::FinishBroadcast(const NodeId& nodeid, bool confirmed_by_node) EXCLUSIVE_LOCKS_REQUIRED(!m_mutex)
```

Comment:
In 4a955955ec587120fb121e5068e3bc1f80406ac5 "net_processing: handle ConnectionType::PRIVATE_BROADCAST connections": It seems `FinishBroadcast` returns a `bool`, but this value is never used.

Reply:
Oh sorry, missed that. It's fine.